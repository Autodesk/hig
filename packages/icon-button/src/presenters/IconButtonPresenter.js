import React, { Component } from "react";
import PropTypes from "prop-types";
import { css } from "emotion";
import ThemeContext from "@hig/theme-context";

import stylesheet from "./stylesheet";
import { AVAILABLE_SURFACES } from "../surfaces";

export default class IconButtonPresenter extends Component {
  static propTypes = {
    disabled: PropTypes.bool,
    hasFocus: PropTypes.bool,
    hasHover: PropTypes.bool,
    icon: PropTypes.element,
    isPressed: PropTypes.bool,
    link: PropTypes.string,
    on: PropTypes.bool,
    onBlur: PropTypes.func,
    onClick: PropTypes.func,
    onFocus: PropTypes.func,
    onMouseDown: PropTypes.func,
    onMouseEnter: PropTypes.func,
    onMouseLeave: PropTypes.func,
    onMouseUp: PropTypes.func,
    surface: PropTypes.oneOf(AVAILABLE_SURFACES),
    title: PropTypes.string.isRequired
  };

  render() {
    const {
      disabled,
      hasFocus,
      hasHover,
      isPressed,
      link,
      on,
      onClick,
      onBlur,
      onFocus,
      onMouseDown,
      onMouseEnter,
      onMouseLeave,
      onMouseUp,
      surface,
      title
    } = this.props;

    return (
      <ThemeContext.Consumer>
        {({ resolvedRoles, metadata }) => {
          const linkProps = link
            ? {
                tabIndex: disabled ? "-1" : "0",
                href: link
              }
            : {};
          const props = {
            ...linkProps,
            disabled,
            hasFocus,
            hasHover,
            isPressed,
            on,
            onClick,
            onBlur,
            onFocus,
            onMouseDown,
            onMouseEnter,
            onMouseLeave,
            onMouseUp,
            surface,
            title
          };
          const Element = this.props.link ? "a" : "button";
          const styles = stylesheet(props, resolvedRoles, metadata.densityId);
          const icon = React.cloneElement(this.props.icon, {
            className: css(styles.iconButtonIcon)
          });
          const tabIndex = disabled ? "-1" : "0";

          return (
            <Element
              className={css(styles.iconButton)}
              disabled={disabled}
              onClick={onClick}
              onBlur={onBlur}
              onFocus={onFocus}
              onMouseDown={onMouseDown}
              onMouseEnter={onMouseEnter}
              onMouseLeave={onMouseLeave}
              onMouseUp={onMouseUp}
              tabIndex={tabIndex}
              title={title}
              {...linkProps}
            >
              {icon}
            </Element>
          );
        }}
      </ThemeContext.Consumer>
    );
  }
}
