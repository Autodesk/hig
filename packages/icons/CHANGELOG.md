# [@hig/icons-v3.4.1](https://github.com/Autodesk/hig/compare/@hig/icons@3.4.0...@hig/icons@3.4.1) (2021-06-26)


### Bug Fixes

* do not spread unknown props to svg ([ee223d1](https://github.com/Autodesk/hig/commit/ee223d1))

# [@hig/icons-v3.4.0](https://github.com/Autodesk/hig/compare/@hig/icons@3.3.0...@hig/icons@3.4.0) (2021-02-22)


### Features

* add stylesheet prop ([decd68a](https://github.com/Autodesk/hig/commit/decd68a))

# [@hig/icons-v3.3.0](https://github.com/Autodesk/hig/compare/@hig/icons@3.2.1...@hig/icons@3.3.0) (2020-05-15)


### Features

* add save-as icons ([197294b](https://github.com/Autodesk/hig/commit/197294b))

# [@hig/icons-v3.2.1](https://github.com/Autodesk/hig/compare/@hig/icons@3.2.0...@hig/icons@3.2.1) (2020-05-12)


### Bug Fixes

* stop using deprecated theme-data roles ([59e91ae](https://github.com/Autodesk/hig/commit/59e91ae))

# [@hig/icons-v3.2.0](https://github.com/Autodesk/hig/compare/@hig/icons@3.1.0...@hig/icons@3.2.0) (2020-05-01)


### Bug Fixes

* remove title element ([8a13814](https://github.com/Autodesk/hig/commit/8a13814))


### Features

* add new icons ([ca0efcd](https://github.com/Autodesk/hig/commit/ca0efcd))

# [@hig/icons-v3.1.0](https://github.com/Autodesk/hig/compare/@hig/icons@3.0.1...@hig/icons@3.1.0) (2019-07-24)


### Features

* distribute SVG source files with [@hig](https://github.com/hig)/icons ([18ac0ed](https://github.com/Autodesk/hig/commit/18ac0ed))

# [@hig/icons-v3.0.1](https://github.com/Autodesk/hig/compare/@hig/icons@3.0.0...@hig/icons@3.0.1) (2019-07-16)


### Bug Fixes

* remove ExternalLinkUI icon ([06fbbde](https://github.com/Autodesk/hig/commit/06fbbde))

# [@hig/icons-v3.0.0](https://github.com/Autodesk/hig/compare/@hig/icons@2.1.2...@hig/icons@3.0.0) (2019-07-09)


### Features

* updating icons ([67a908a](https://github.com/Autodesk/hig/commit/67a908a))


### BREAKING CHANGES

* Updated styling of all 16 and 24px icons to better align with existing command icons

* Added add-page
* Added external-link
* Added file-audio
* Added import
* Added inbox
* Added mute
* Added not-editable
* Added select-all
* Added select-none
* Added sort
* Added unlink
* Added unpin
* Added user-inactive
* Renamed building-ops to building
* Renamed cost-control to bill-stack
* Renamed export to export-move
* Renamed insight to graph
* Renamed issue to alert
* Renamed layout to survey-equipment
* Renamed user-incactive to user-inactive
* Removed assets
* Removed caret - can be found in UI Controls
* Removed check-disabled - can be found in UI Controls
* Removed check-white
* Removed checkbox-checked - can be found in UI Controls
* Removed checkbox-indeterminate - can be found in UI Controls
* Removed checkmark-blue-dark
* Removed checkmark-indeterminate - can be found in UI Controls
* Removed checkmark-s - can be found in UI Controls
* Removed clear-small
* Removed close - can be found in UI Controls
* Removed close-hamburger - can be found in UI Controls
* Removed close-notification - can be found in UI Controls
* Removed close-xs - can be found in UI Controls
* Removed dropdown-caret-down - can be found in UI Controls
* Removed dropdown-caret-left - can be found in UI Controls
* Removed dropdown-caret-right - can be found in UI Controls
* Removed dropdown-caret-up - can be found in UI Controls
* Removed hamburger - can be found in UI Controls
* Removed hamburger-close - can be found in UI Controls
* Removed hamburger-default - can be found in UI Controls
* Removed model-coordination
* Removed rfi
* Removed social-blog-2
* Removed x-close-gray

# [@hig/icons-v2.1.2](https://github.com/Autodesk/hig/compare/@hig/icons@2.1.1...@hig/icons@2.1.2) (2019-06-25)


### Bug Fixes

* don't show tooltip on hover ([e50e34f](https://github.com/Autodesk/hig/commit/e50e34f))

# [@hig/icons-v2.1.1](https://github.com/Autodesk/hig/compare/@hig/icons@2.1.0...@hig/icons@2.1.1) (2019-03-13)


### Bug Fixes

* theme-context and theme-data as peer dependencies ([34cd334](https://github.com/Autodesk/hig/commit/34cd334))

# [@hig/icons-v2.1.0](https://github.com/Autodesk/hig/compare/@hig/icons@2.0.2...@hig/icons@2.1.0) (2019-03-08)


### Features

* add ui control icons ([5465341](https://github.com/Autodesk/hig/commit/5465341))

# [@hig/icons-v2.0.2](https://github.com/Autodesk/hig/compare/@hig/icons@2.0.1...@hig/icons@2.0.2) (2019-02-08)


### Bug Fixes

* bump up theme-context dependency ([b068dab](https://github.com/Autodesk/hig/commit/b068dab))

# [@hig/icons-v2.0.1](https://github.com/Autodesk/hig/compare/@hig/icons@2.0.0...@hig/icons@2.0.1) (2019-01-23)


### Bug Fixes

* **package:** update [@hig](https://github.com/hig)/theme-context to version 2.0.0 ([7975948](https://github.com/Autodesk/hig/commit/7975948))

# [@hig/icons-v2.0.0](https://github.com/Autodesk/hig/compare/@hig/icons@1.0.0...@hig/icons@2.0.0) (2019-01-17)


### chore

* **icons:** release icons 2.0.0 ([b61b2cb](https://github.com/Autodesk/hig/commit/b61b2cb))


### Features

* **icons:** make icons themable ([56abadd](https://github.com/Autodesk/hig/commit/56abadd))
* support color attribute on avatar ([c7121d1](https://github.com/Autodesk/hig/commit/c7121d1))


### BREAKING CHANGES

* **icons:** update package version to release icons 2.0.0

# [@hig/icons-v1.0.0](https://github.com/Autodesk/hig/compare/@hig/icons@0.2.1...@hig/icons@1.0.0) (2018-12-04)


### BREAKING

* **icon/icon-button:** Removes [@hig](https://github.com/hig)/icon and refactor [@hig](https://github.com/hig)/icon-button to use [@hig](https://github.com/hig)/icons ([49d78f4](https://github.com/Autodesk/hig/commit/49d78f4))


### Bug Fixes

* **storybook:** Add back Icon stories ([f38f2d4](https://github.com/Autodesk/hig/commit/f38f2d4))


### BREAKING CHANGES

* **icon/icon-button:** Removes @hig/icon
* **icon/icon-button:** Use icon prop instead of name or svg in @hig/icon-button to render Icon/svg
* **icon/icon-button:** No longer export `names` from @hig/icon-button

Usage:
```
import { Assets24 } from "@hig/icons"

function MyComponent() {
  return (
    <IconButton
      icon={<Assets24 />}
      onClick={() => console.log('Clicked it.')}
    />
  );
}
```

# [@hig/icons-v0.2.1](https://github.com/Autodesk/hig/compare/@hig/icons@0.2.0...@hig/icons@0.2.1) (2018-10-08)


### Bug Fixes

* replace stroke icons with fill ([5e06c82](https://github.com/Autodesk/hig/commit/5e06c82))

# [@hig/icons-v0.2.0](https://github.com/Autodesk/hig/compare/@hig/icons@0.1.0...@hig/icons@0.2.0) (2018-08-07)


### Features

* update icon set ([80bf8c6](https://github.com/Autodesk/hig/commit/80bf8c6))
