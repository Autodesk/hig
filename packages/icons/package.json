{
  "name": "@hig/icons",
  "version": "0.1.0",
  "description": "HIG icon assets",
  "author": "Autodesk Inc.",
  "license": "Apache-2.0",
  "homepage": "https://hig.autodesk.com",
  "repository": {
    "type": "git",
    "url": "https://github.com/Autodesk/hig.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "main": "build/index.js",
  "module": "build/index.es.js",
  "files": [
    "build/*"
  ],
  "devDependencies": {
    "@hig/babel-preset": "^0.1.0",
    "@hig/eslint-config": "^0.1.0",
    "@hig/jest-preset": "^0.1.0",
    "@hig/scripts": "^0.1.2",
    "@hig/semantic-release-config": "^0.1.0",
    "@hig/styles": "^0.1.1",
    "svgo": "^1.0.5"
  },
  "peerDependencies": {
    "react": "^15.4.1 || ^16.3.2"
  },
  "scripts": {
    "build": "hig-scripts-build",
    "build-icons": "node scripts/build-icons/index.js"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  },
  "release": {
    "extends": "@hig/semantic-release-config"
  },
  "babel": {
    "env": {
      "test": {
        "presets": [
          "@hig/babel-preset/test"
        ]
      }
    }
  }
}
