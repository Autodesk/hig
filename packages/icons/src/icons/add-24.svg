<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
  <title>add-24</title>
  <g id="final">
    <g id="add">
      <path id="outline" d="M20,10H14V4a1,1,0,0,0-1-1H11a1,1,0,0,0-1,1v6H4a1,1,0,0,0-1,1v2a1,1,0,0,0,1,1h6.13v6a1,1,0,0,0,1,1H13a1,1,0,0,0,1-1V14h6a1,1,0,0,0,1-1V11A1,1,0,0,0,20,10Zm0,3H13v7H11V13H4V11h7V4h2v7h7Z" fill="#2e4258"/>
    </g>
  </g>
</svg>
