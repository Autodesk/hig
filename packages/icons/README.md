# Icons

This package retains the source of the library of HIG icons.
You can import individual icons to render on your app.

```jsx
import { Assets24 } from "@hig/icons";

const MyComponent = () => (
    <div className="my-class">
        <Assets24 />
    </div>
);

export default MyComponent
```

Read more about where and how to use icons on [the website](https://hig.autodesk.com/web/basics/icons).

## Getting started

### Install the package

```bash
yarn add @hig/icons @hig/theme-context @hig/theme-data
```

# Developing

## Adding a New Icon

To add a new SVG icon to the library, add the `svg` file to the `src/icons` directory.

Then run the following command:

```bash
yarn build-icons
```

This will use SVGO to optimize the new icon, and update our release files. Remember to commit these changes.
