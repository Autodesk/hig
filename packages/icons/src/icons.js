import "./icons.scss";
import icons from "./release/hig-icons-bundle";
import names from "./release/hig-icon-names";

const AVAILABLE_NAMES = Object.freeze(Object.values(names));

export { icons as default, AVAILABLE_NAMES, names };
