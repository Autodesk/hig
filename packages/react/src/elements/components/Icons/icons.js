import "./icons.scss";

export { default } from "./release/hig-icons-bundle";
export { default as names } from "./release/hig-icon-names";
