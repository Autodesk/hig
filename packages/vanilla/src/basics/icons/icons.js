import './icons.scss';

var IconBundle = require("./release/hig-icons-bundle.js");

module.exports = IconBundle;