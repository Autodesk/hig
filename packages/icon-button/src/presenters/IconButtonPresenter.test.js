import renderer from "react-test-renderer";
import React from "react";

import { Settings24 } from "@hig/icons";
import IconButtonPresenter from "./IconButtonPresenter";

describe("IconButtonPresenter", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<IconButtonPresenter title="Settings" icon={<Settings24 />} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with a link", () => {
    const tree = renderer
      .create(
        <IconButtonPresenter
          title="Settings"
          link="//example.com"
          icon={<Settings24 />}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when disabled", () => {
    const tree = renderer
      .create(
        <IconButtonPresenter
          disabled
          title="Settings"
          link="//example.com"
          icon={<Settings24 />}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
