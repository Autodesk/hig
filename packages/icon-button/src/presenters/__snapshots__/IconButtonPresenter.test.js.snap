// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtonPresenter renders correctly 1`] = `
.emotion-1 {
  border: 2px solid transparent;
  display: inline-block;
  position: relative;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 2px;
  padding: 0;
  height: calc(20px + (8px * 2));
  line-height: calc(20px + (8px * 2));
  width: calc(20px + (8px * 2));
  outline: 0;
  -webkit-transition-property: box-shadow,background-color;
  transition-property: box-shadow,background-color;
  -webkit-transition-duration: 0.3s,0.3s;
  transition-duration: 0.3s,0.3s;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  box-shadow: none;
  background-color: transparent;
}

.emotion-1 svg * {
  fill: #808080;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: fill;
  transition-property: fill;
}

.emotion-0 {
  fill: #808080;
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  margin: 0 auto;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  pointer-events: none;
}

.emotion-0 > * {
  fill: #808080;
  pointer-events: none;
}

<button
  className="emotion-1"
  disabled={undefined}
  onBlur={undefined}
  onClick={undefined}
  onFocus={undefined}
  onMouseDown={undefined}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  onMouseUp={undefined}
  tabIndex="0"
  title="Settings"
>
  <svg
    className="emotion-0"
    height="24px"
    viewBox="0 0 24 24"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      settings-24
    </title>
    <path
      d="M12 7.67A4.33 4.33 0 1 0 16.33 12 4.34 4.34 0 0 0 12 7.67zm0 7.66A3.33 3.33 0 1 1 15.33 12 3.33 3.33 0 0 1 12 15.33z"
    />
    <path
      d="M22 14v-4h-2.95a7.22 7.22 0 0 0-.65-1.57l2.09-2.09-2.83-2.83-2.09 2.09A7.22 7.22 0 0 0 14 5V2h-4v3a7.22 7.22 0 0 0-1.57.65L6.34 3.51 3.51 6.34 5.6 8.43A7.22 7.22 0 0 0 5 10H2v4h3a7.22 7.22 0 0 0 .65 1.57l-2.14 2.09 2.83 2.83 2.09-2.09a7.22 7.22 0 0 0 1.57.65V22h4v-2.95a7.22 7.22 0 0 0 1.57-.65l2.09 2.09 2.83-2.83-2.09-2.09a7.22 7.22 0 0 0 .65-1.57zm-2.93 3.66l-1.41 1.41-1.91-1.91-.66.37a7.07 7.07 0 0 1-1.36.56l-.73.2V21h-2v-2.71l-.73-.2a7.07 7.07 0 0 1-1.36-.56l-.66-.37-1.91 1.91-1.41-1.41 1.91-1.92-.37-.66a6.73 6.73 0 0 1-.56-1.35l-.2-.73H3v-2h2.71l.2-.73a6.73 6.73 0 0 1 .56-1.35l.37-.66-1.91-1.92 1.41-1.41 1.91 1.91.66-.37a7.07 7.07 0 0 1 1.36-.56l.73-.2V3h2v2.71l.73.2a7.07 7.07 0 0 1 1.36.56l.66.37 1.91-1.91 1.41 1.41-1.91 1.92.37.66a6.73 6.73 0 0 1 .56 1.35l.2.73H21v2h-2.71l-.2.73a6.73 6.73 0 0 1-.56 1.35l-.37.66z"
    />
  </svg>
</button>
`;

exports[`IconButtonPresenter renders correctly when disabled 1`] = `
.emotion-0 {
  fill: #808080;
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  margin: 0 auto;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  pointer-events: none;
}

.emotion-0 > * {
  fill: #808080;
  pointer-events: none;
}

.emotion-1 {
  border: 2px solid transparent;
  display: inline-block;
  position: relative;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 2px;
  padding: 0;
  height: calc(20px + (8px * 2));
  line-height: calc(20px + (8px * 2));
  width: calc(20px + (8px * 2));
  outline: 0;
  -webkit-transition-property: box-shadow,background-color;
  transition-property: box-shadow,background-color;
  -webkit-transition-duration: 0.3s,0.3s;
  transition-duration: 0.3s,0.3s;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  box-shadow: none;
  background-color: transparent;
  opacity: 0.4;
  pointer-events: none;
}

.emotion-1 svg * {
  fill: #808080;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: fill;
  transition-property: fill;
}

<a
  className="emotion-1"
  disabled={true}
  href="//example.com"
  onBlur={undefined}
  onClick={undefined}
  onFocus={undefined}
  onMouseDown={undefined}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  onMouseUp={undefined}
  tabIndex="-1"
  title="Settings"
>
  <svg
    className="emotion-0"
    height="24px"
    viewBox="0 0 24 24"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      settings-24
    </title>
    <path
      d="M12 7.67A4.33 4.33 0 1 0 16.33 12 4.34 4.34 0 0 0 12 7.67zm0 7.66A3.33 3.33 0 1 1 15.33 12 3.33 3.33 0 0 1 12 15.33z"
    />
    <path
      d="M22 14v-4h-2.95a7.22 7.22 0 0 0-.65-1.57l2.09-2.09-2.83-2.83-2.09 2.09A7.22 7.22 0 0 0 14 5V2h-4v3a7.22 7.22 0 0 0-1.57.65L6.34 3.51 3.51 6.34 5.6 8.43A7.22 7.22 0 0 0 5 10H2v4h3a7.22 7.22 0 0 0 .65 1.57l-2.14 2.09 2.83 2.83 2.09-2.09a7.22 7.22 0 0 0 1.57.65V22h4v-2.95a7.22 7.22 0 0 0 1.57-.65l2.09 2.09 2.83-2.83-2.09-2.09a7.22 7.22 0 0 0 .65-1.57zm-2.93 3.66l-1.41 1.41-1.91-1.91-.66.37a7.07 7.07 0 0 1-1.36.56l-.73.2V21h-2v-2.71l-.73-.2a7.07 7.07 0 0 1-1.36-.56l-.66-.37-1.91 1.91-1.41-1.41 1.91-1.92-.37-.66a6.73 6.73 0 0 1-.56-1.35l-.2-.73H3v-2h2.71l.2-.73a6.73 6.73 0 0 1 .56-1.35l.37-.66-1.91-1.92 1.41-1.41 1.91 1.91.66-.37a7.07 7.07 0 0 1 1.36-.56l.73-.2V3h2v2.71l.73.2a7.07 7.07 0 0 1 1.36.56l.66.37 1.91-1.91 1.41 1.41-1.91 1.92.37.66a6.73 6.73 0 0 1 .56 1.35l.2.73H21v2h-2.71l-.2.73a6.73 6.73 0 0 1-.56 1.35l-.37.66z"
    />
  </svg>
</a>
`;

exports[`IconButtonPresenter renders correctly with a link 1`] = `
.emotion-1 {
  border: 2px solid transparent;
  display: inline-block;
  position: relative;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 2px;
  padding: 0;
  height: calc(20px + (8px * 2));
  line-height: calc(20px + (8px * 2));
  width: calc(20px + (8px * 2));
  outline: 0;
  -webkit-transition-property: box-shadow,background-color;
  transition-property: box-shadow,background-color;
  -webkit-transition-duration: 0.3s,0.3s;
  transition-duration: 0.3s,0.3s;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  box-shadow: none;
  background-color: transparent;
}

.emotion-1 svg * {
  fill: #808080;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: fill;
  transition-property: fill;
}

.emotion-0 {
  fill: #808080;
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  margin: 0 auto;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  pointer-events: none;
}

.emotion-0 > * {
  fill: #808080;
  pointer-events: none;
}

<a
  className="emotion-1"
  disabled={undefined}
  href="//example.com"
  onBlur={undefined}
  onClick={undefined}
  onFocus={undefined}
  onMouseDown={undefined}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  onMouseUp={undefined}
  tabIndex="0"
  title="Settings"
>
  <svg
    className="emotion-0"
    height="24px"
    viewBox="0 0 24 24"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      settings-24
    </title>
    <path
      d="M12 7.67A4.33 4.33 0 1 0 16.33 12 4.34 4.34 0 0 0 12 7.67zm0 7.66A3.33 3.33 0 1 1 15.33 12 3.33 3.33 0 0 1 12 15.33z"
    />
    <path
      d="M22 14v-4h-2.95a7.22 7.22 0 0 0-.65-1.57l2.09-2.09-2.83-2.83-2.09 2.09A7.22 7.22 0 0 0 14 5V2h-4v3a7.22 7.22 0 0 0-1.57.65L6.34 3.51 3.51 6.34 5.6 8.43A7.22 7.22 0 0 0 5 10H2v4h3a7.22 7.22 0 0 0 .65 1.57l-2.14 2.09 2.83 2.83 2.09-2.09a7.22 7.22 0 0 0 1.57.65V22h4v-2.95a7.22 7.22 0 0 0 1.57-.65l2.09 2.09 2.83-2.83-2.09-2.09a7.22 7.22 0 0 0 .65-1.57zm-2.93 3.66l-1.41 1.41-1.91-1.91-.66.37a7.07 7.07 0 0 1-1.36.56l-.73.2V21h-2v-2.71l-.73-.2a7.07 7.07 0 0 1-1.36-.56l-.66-.37-1.91 1.91-1.41-1.41 1.91-1.92-.37-.66a6.73 6.73 0 0 1-.56-1.35l-.2-.73H3v-2h2.71l.2-.73a6.73 6.73 0 0 1 .56-1.35l.37-.66-1.91-1.92 1.41-1.41 1.91 1.91.66-.37a7.07 7.07 0 0 1 1.36-.56l.73-.2V3h2v2.71l.73.2a7.07 7.07 0 0 1 1.36.56l.66.37 1.91-1.91 1.41 1.41-1.91 1.92.37.66a6.73 6.73 0 0 1 .56 1.35l.2.73H21v2h-2.71l-.2.73a6.73 6.73 0 0 1-.56 1.35l-.37.66z"
    />
  </svg>
</a>
`;
